pytorch_lightning/trainer/evaluation_loop_mixin.py

```diff
@@ -110,6 +114,21 @@ def run_evaluation(self, test=False):
             if self.fast_dev_run:
                 max_batches = 1
 
+            # init validation or test progress bar
+            if test:
+                # main progress bar will already be closed when testing so initial position is free
+                pbar = tqdm.tqdm(desc='Testing', total=max_batches,
+                                 leave=True, position=2 * self.process_position,
+                                 disable=not self.show_progress_bar, dynamic_ncols=True,
+                                 unit='batch')
+                self.test_progress_bar = pbar
+            else:
+                pbar = tqdm.tqdm(desc='Validating', total=max_batches,
+                                 leave=False, position=2 * self.process_position + 1,
```

Comment:
Because initial position is occupied by the main progress bar when we validating. And when we testing there is no active progress bar so we can use initial position.

Reply:
Sorry, I don't understand how the bar activity is connected to the size + 1...

Reply:
Well, let me try to explain. How roughly tqdm works. If you have only one progress bar tqdm uses carriage return (`\r`) for overwriting this progress bar. Even in this simple case there may be problems. If someone print something using `print()` it will move global cursor and tqdm will not know about that. So progress bar will now occupy some new line and we will have remnants of its old state. Something like that:
```
30%|█████████████████████████▏                                                          | 3/10 [00:03<00:07,  1.00s/it]
Something                                                                                                                
60%|██████████████████████████████████████████████████▍                                 | 6/10 [00:06<00:04,  1.00s/it]
```

And when there are several progress bars the things get more complicated. From tqdm docs:
>position : int, optional
>Specify the line offset to print this bar (starting from 0) Automatic if unspecified. Useful to manage multiple bars at once (eg, from threads).

So tqdm should remember relative line numbers of each active progress bar (because it should know what line to update). Tqdm can make it automatically but not in multithreading mode. So we should manually set position of each progress bar.

Going back to our case, we have at most 2 progress bars at once. And it happens only while validating in main train loop. Previously there was only one progress bar and its position was equal to `self.process_position`. But as I have already said now there can be two progress bar an once. So I decided to define main position as `2 * self.process_position` and define auxiliary position as `2 * self.process_position + 1`.

So, when we make validation sanity check or testing, we just use main position. When we are training, we use main position for the main progress bar with total number of batches, and auxiliary position for validating progress bar.

Also you can just remove this +1 and look what happens for progress bars in the example code :)

Reply:
I think that I see your point... I am not just much happy about this hacking solution, I would personally think about another bar without this kind of issue... but if we want to stay with TQDM, then it is probably the only way... :(

Reply:
Yeah, I agree that tqdm has some unpleasant issues. At least it would be very desired to change all `print()` to `tqdm.write()` in order to not break progress bars by outer prints... I had a look to enlighten and it indeed looks interesting. But if I understand right there is no way to use it in multithreading mode :(

But actually I am not sure that I understand well when we may want to train different models from one terminal.